UserStudentApp/views.py

```diff
@@ -0,0 +1,87 @@
+from .models import Student, CustomUser, Course, UserImage
+from .serializers import StudentSerializer, UserSerializer, CourseSerializer, ImageSerializer
+from rest_framework import permissions,viewsets,status
+from .permissions import IsOwnerOrReadOnly
+from rest_framework.response import Response
+from django.db.models import Q
+
+
+class UserViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    This viewset automatically provides `list` and `retrieve` actions.
+    """
+    permission_classes  = [IsOwnerOrReadOnly,permissions.IsAuthenticated]
+    serializer_class    = UserSerializer
+
+
+    def get_queryset(self):
+        request_user = self.request.user
+        name         = self.request.query_params.get('name')
+        
+        if request_user.is_superuser:
+            queryset = CustomUser.objects.all()
+            if name:
+                queryset = CustomUser.objects.filter(username= name) 
+        else:
+            queryset = CustomUser.objects.filter(email= request_user)
+
+        return queryset
+  
+
+class StudentViewSet(viewsets.ModelViewSet):
+    """
+    This viewset automatically provides `list`, `create`, `retrieve`,
+    `update` and `destroy` actions.
+    """
+
+    permission_classes = [IsOwnerOrReadOnly,permissions.IsAuthenticated]
+    serializer_class   = StudentSerializer
+
+
+    def get_queryset(self, *args, **kwargs):
+        request_user = self.request.user
+        if request_user.is_superuser:
+            queryset = Student.objects.all()
+            names    = self.request.query_params.getlist('name')
+            if names:
+                queryset = Student.objects.filter(Q(name__in= names) | Q(owner__username__in= names))
+        else:
+            queryset = Student.objects.filter(owner = request_user)
+            name     = self.request.query_params.get('name')
+            age      = self.request.query_params.get('age')
+            if name and age:
```

Comment:
i don't think line#52 and 53 are required.

Reply:
i know that scenario. Please find out a solution to do it without the first if condition.